pkgs/tools/networking/v2ray/generic.nix

```diff
@@ -0,0 +1,53 @@
+{ lib, linkFarm, buildGoPackage, runCommand, makeWrapper, go
+
+# Version specific args
+, version, src, assets
+, ... }:
+
+let
+  assetsDrv = linkFarm "v2ray-assets" (lib.mapAttrsToList (name: path: {
+    inherit name path;
+  }) assets);
+
+  core = buildGoPackage rec {
+    pname = "v2ray-core";
+    inherit version src;
+
+    goPackagePath = "v2ray.com/core";
+    goDeps = ./deps.nix;
+
+    nativeBuildInputs = [ go ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      go build -o v2ray v2ray.com/core/main
+      go build -o v2ctl v2ray.com/core/infra/control/main
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      install -Dm755 v2ray v2ctl -t $bin/bin
+    '';
+  };
+
+in runCommand "v2ray-${version}" {
```

Comment:
I feel like this could be put into a `postFixup` phase.

@kalbasit for his go knowledge

Reply:
I make this separated to avoid rebuilding the core when assets updated. Assets are data files (like builtin plugins) and I think they should more extensible.

Reply:
that's fair... just seems like a lot going on within a single file. But I don't feel super strongly either way.